extensions-core/multi-stage-query/src/test/java/org/apache/druid/msq/test/MSQTestBase.java

```diff
@@ -385,10 +386,11 @@ public String getFormatString()
     objectMapper = setupObjectMapper(injector);
     objectMapper.registerModules(sqlModule.getJacksonModules());
 
+    testTaskActionClient = Mockito.spy(new MSQTestTaskActionClient(objectMapper));
```

Comment:
Is there a reason we are spying on this object ?

Reply:
Yes, to check for exceptions where segments cannot be allocated, like InsertCannotAllocateSegmentFault